docs/wiki/[Variables]-configure_connectivity_resources.md

```diff
@@ -1 +1,545 @@
+                gateway_sku_expressroute = "ErGw2AZ"
+                gateway_sku_vpn          = "VpnGw3"
+              }
+            }
+            azure_firewall = {
+              enabled = false
+              config = {
+                address_prefix   = "10.100.0.0/24"
+                enable_dns_proxy = true
+                availability_zones = {
+                  zone_1 = true
+                  zone_2 = true
+                  zone_3 = true
+                }
+              }
+            }
+            spoke_virtual_network_resource_ids      = []
+            enable_outbound_virtual_network_peering = false
+          }
+        },
+      ]
+```
+
+Each entry in the `hub_network` list will create a new set of resources based on the configured values.
+Each entry includes an `enable` (`bool`) control to allow the configuration to be enabled and disabled, and a `config` (`object`) which contains the configuration settings for each hub network.
+
+> **IMPORTANT**: When creating multiple hub networks, each must use a uniquely defined `location`.
+
+#### `settings.hub_networks[].enable`
+
+The `enable` (`bool`) input allows you to toggle whether to create this hub network instance, including all associated resources. Set to `false` if you want to toggle individual hub network instances without removing the full configuration.
+
+#### `settings.hub_networks[].config`
+
+The `config` (`object`) input allows you to set the following configuration items for each hub network:
+
+##### `settings.hub_networks[].config.address_space`
+
+List of IP Address ranges in CIDR (prefix) notation for the Virtual Network.
+Must be a valid entry for a Virtual Network `address_space` input.
+
+##### `settings.hub_networks[].config.location`
+
+Set the location/region where the hub network and associated resources are created.
+Changing this forces new resources to be created.
+By default, a `hub_network` with an empty value in the `location` field will be deployed to the location inherited from either `configure_connectivity_resources.location`, or the top-level variable `default_location`, in order of precedence.
+
+##### `settings.hub_networks[].config.link_to_ddos_protection_plan`
+
+The `link_to_ddos_protection_plan` (`bool`) input controls whether to link the hub network to the DDoS Protection plan managed by the module.
+
+For this to work, the `ddos_protection_plan.enabled` value must be set to `true`.
+
+##### `settings.hub_networks[].config.dns_servers`
+
+Specify a list of custom DNS server IP addresses.
+
+##### `settings.hub_networks[].config.bgp_community`
+
+Specify the BGP community attribute for the Virtual Network in format `<as-number>:<community-value>`.
+
+##### `settings.hub_networks[].config.subnets`
+
+List of subnet configuration objects, used to extend the Virtual Network with custom subnets.
+
+Each entry within this list must contain the following configuration `object()`:
+
+```hcl
+object({
```

Comment:
Could we provide an example here of multiple subnets in a list(object({})). These more complex data structures could do will illustration IMHO.

e.g.

```hcl
[
  {
    name = "mysubnet"
    address_prefixes = [
      "192.168.1.0/24",
      "192.168.2.0/24
    ]
    network_security_group_id = ""
    route_table_id = ""
  },
  {
    name = "mysubnet2"
    address_prefixes = [
      "192.168.3.0/24",
      "192.168.4.0/24
    ]
    network_security_group_id = ""
    route_table_id = ""
  }
]
```

Reply:
Absolutely... this would be a great addition. We just need to agree whether this is the right place, or whether we need to consider breaking this (already huge) doc into smaller pieces which are easier to consume?